keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/BakeryLifecycleMonitor.kt

```diff
@@ -71,6 +85,51 @@ class BakeryLifecycleMonitor(
       }
   }
 
+  /**
+   * Stores the ami details from a successful bake task so we don't have to wait for those
+   * details to appear in the clouddriver cache (~5 minute delay every time).
+   *
+   * If there's a problem parsing we log an error and move on, because we will see
+   * the image in clouddriver eventually.
+   */
+  suspend fun storeBakedImage(execution: ExecutionDetailResponse) {
+    // we know we launch a bake task with a stage called "bake".
+    val bakeStageRaw = execution.execution.stages?.find { it["type"] == "bake" }
+    if (bakeStageRaw == null) {
+      log.error("Trying to find baked ami information, but can't find a bake stage in execution ${execution.id}")
+      return
+    }
+    try {
+      val bakeStage = objectMapper.convertValue<BakeStage>(bakeStageRaw)
+      val details = bakeStage.outputs.deploymentDetails
+      if (details.isEmpty()) {
+        log.error("No bake details in the bake stage in execution ${execution.id}")
+      }
+
+      // use the first region present because the only difference should be in ami id and region name
+      // even the base ami app version will be the same across regions
+      val bakedImage = BakedImage(
+        name = details.first().imageName,
```

Comment:
Since you're using `details.first` everywhere, you could replace lines 104-105 with the following to make the code a bit cleaner:
```kotlin
val detail = bakeStage.outputs.deploymentDetails.firstOrNull()
if (detail == null) {
```

Reply:
or use a `with` block

Reply:
good call I'll do that